### ISSUE TITLE
OSS Architecture: Combining front-end and back-end into one repo

### ISSUE THREAD

**Issue by mahrsbar** [ProgramEquity/amplify-back-end#127]
Goal: Combining front end and back end into one repo that deploys to a staging build (for hackathon) and production build (for Climate can't Wait starting Jan 8th)

**Why are we doing this?**
- Allows us to use CodeSpace more efficiently
- Alleviates security concerns from using Cors

**Notes from front-end:**
- We're using Vue 3.6.2 for the framework. These components will be need to be migrated over.
- Front-end is using [back-end apis](https://github.com/ProgramEquity/amplify-back-end/wiki) but unsure if this will be affected by migration
- Need to incorporate LESS and other libraries?
- Need to hook express into Vue

Feedback wanted on the following @JamesMGreene @mscoutermarsh @wallace @BeeXiong  @teakopp :
- Discussion on 11/30 we theorized all frontend calls will remain the same 
- How will other app routing need to be changed? 
- Are all frontend files going under a public or src file within backend 
- Are there major changes to how backend and frontend components are stored?
- Is it correct that node will now have a production build vs a staging one

**Comment by manishapriya94** [comment]
Conversation with mike: https://github.zoom.us/rec/share/lPFDoaBMpjbESPjCvkzVmZpohDH_q_phs0PiegKIrsoMklg6dj6GEaWKcJ1lukTb.G4nJTWuOh5kmpeMx Passcode: NjXWp7U#

Seems it might be best for us to go back to a single app structure, for [reference](https://github.com/ProgramEquity/ally-guide-app)

**Comment by JamesMGreene** [comment]
Sorry for taking so long to post here.

Back when I was researching for this, I found that it is easy enough to have an Express app host a **built** production Vue app but I struggled to find good resources on how to do it in a way that still supported hot-reloading of Vue from a development perspective.

The most interesting resource I found on the topic was [this one](https://dennisreimann.de/articles/vue-cli-serve-express.html), which seemed very promising in its unique approach of mounting the Express app into the Vue app (`vue-cli-service` app, that is) since it already has an internal Express server. 🔄
